src/lib/mdns/minimal/tests/CheckOnlyServer.h

```diff
@@ -36,85 +38,188 @@ namespace mdns {
 namespace Minimal {
 namespace test {
 
-class CheckOnlyServer : public ServerBase, public ParserDelegate
+constexpr QNamePart kIgnoreQNameParts[] = { "IGNORE", "THIS" };
+namespace {
+bool StringMatches(const BytesRange & br, const char * str)
+{
+    return br.Size() == strlen(str) && strncmp(str, reinterpret_cast<const char *>(br.Start()), br.Size()) == 0;
+}
+
+template <size_t N>
+void MakePrintableName(char (&location)[N], SerializedQNameIterator name)
+{
+    auto buf = chip::Encoding::BigEndian::BufferWriter(reinterpret_cast<uint8_t *>(&location[0]), N);
+    while (name.Next())
+    {
+        buf.Put(name.Value());
+        buf.Put(".");
+    }
+    buf.Put('\0');
+}
+
+template <size_t N>
+void MakePrintableName(char (&location)[N], FullQName name)
+{
+    auto buf = chip::Encoding::BigEndian::BufferWriter(reinterpret_cast<uint8_t *>(&location[0]), N);
+    for (size_t i = 0; i < name.nameCount; ++i)
+    {
+        buf.Put(name.names[i]);
+        buf.Put(".");
+    }
+    buf.Put('\0');
+}
+#define CHECK_CONDITION(x)                                                                                                         \
+    if (mInSuite != nullptr)                                                                                                       \
```

Comment:
add a comment on why mInSuite check is needed. What path will do the CHECK without an active test?

Reply:
So this was originally from the initial build of the check server, when I was inserting the CheckServer at compile time, before the suite was generated. The default constructor had a null suite, so the check was required in case someone writing a test (ie, me) forgot to set the suite before running the tests. But actually, now that you point it out, the default constructor isn't really needed anymore so I can remove this check completely. Done in next commit.